contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,325 @@
+            .rawValue;
+
+        collateralReturned = longCollateralRedeemed + shortCollateralRedeemed;
+        collateralToken.safeTransfer(msg.sender, collateralReturned);
+
+        emit PositionSettled(msg.sender, collateralReturned, longTokensToRedeem, shortTokensToRedeem);
+    }
+
+    /****************************************
+     *          INTERNAL FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Request a price in the optimistic oracle for a given request timestamp and ancillary data combo. Set the bonds
+     * accordingly to the deployer's parameters. Will revert if re-requesting for a previously requested combo.
+     */
+    function _requestOraclePrice() internal {
+        OptimisticOracleInterface optimisticOracle = _getOO();
+
+        collateralToken.safeApprove(address(optimisticOracle), proposerReward);
+
+        optimisticOracle.requestPrice(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            collateralToken,
+            proposerReward
+        );
+
+        // Set the Optimistic oracle liveness for the price request.
+        optimisticOracle.setCustomLiveness(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleLivenessTime
+        );
+
+        // Set the Optimistic oracle proposer bond for the price request.
+        optimisticOracle.setBond(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleProposerBond
+        );
+
+        // Make the request an event-based request.
+        OptimisticOracleInterfaceEventBased(address(optimisticOracle)).setEventBased(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData
+        );
+
+        priceRequested = true;
+    }
+
+    /**
+     * @notice Returns a number between 0 and 1e18 to indicate how much collateral each long and short token are entitled
+     * to per collateralPerPair.
+     * @param _expiryPrice price from the optimistic oracle for the LSP price identifier.
+     * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
+     */
+    function percentageLongCollAtExpiry(int256 _expiryPrice) internal view returns (uint256) {
+        if (_expiryPrice >= strikePrice) return FixedPoint.fromUnscaledUint(1).rawValue;
+        else return FixedPoint.fromUnscaledUint(0).rawValue;
+    }
+
+    /**
+     * @notice Fetch the optimistic oracle expiration price. If the oracle has the price for the provided expiration timestamp
+     * and customData combo then store this. If there is no price revert.
+     */
+    function getExpirationPrice() internal hasPrice {
+        expiryPrice = _getOP(expirationTimestamp, customAncillaryData);
```

Comment:
can you re-name this `_getOP` function to `getOraclePrice`? prehaps? sorry...I know I asked you in a previous PR to make things shorter but this is a bit confusing